misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java

```diff
@@ -564,24 +564,31 @@ private void executeFunction(Test test, TestSuite suite, String packageName, Cla
-                    List<Object[]> argList = extractArguments((BMap) valueSets);
-                    int i = 0;
-                    for (Object[] arg : argList) {
-                        invokeDataDrivenTest(suite, test.getTestName(), escapeSpecialCharacters(keyValues.get(i)),
-                                classLoader, scheduler, shouldSkip, packageName, arg, argTypes, failedOrSkippedTests);
-                        i++;
+                    if (((BMap) valueSets).isEmpty()) {
+                        computeFunctionResult(test.getTestName(), packageName, shouldSkip, failedOrSkippedTests,
+                                new Error("The provided data set is empty."));
+                    } else {
+                        List<String> keyValues = getKeyValues((BMap) valueSets);
+                        Class<?>[] argTypes = extractArgumentTypes((BMap) valueSets);
+                        List<Object[]> argList = extractArguments((BMap) valueSets);
+                        for (int i = 0, argListSize = argList.size(); i < argListSize; i++) {
+                            invokeDataDrivenTest(suite, test.getTestName(), escapeSpecialCharacters(keyValues.get(i)),
+                                    classLoader, scheduler, shouldSkip, packageName, argList.get(i), argTypes,
+                                    failedOrSkippedTests);
+                        }
                     }
                 } else if (valueSets instanceof BArray) {
-                    // Handle array data sets
-                    Class<?>[] argTypes = extractArgumentTypes((BArray) valueSets);
-                    List<Object[]> argList = extractArguments((BArray) valueSets);
-                    int i = 0;
-                    for (Object[] arg : argList) {
-                        invokeDataDrivenTest(suite, test.getTestName(), String.valueOf(i), classLoader, scheduler,
-                                shouldSkip, packageName, arg, argTypes, failedOrSkippedTests);
-                        i++;
+                    if (((BArray) valueSets).isEmpty()) {
+                        computeFunctionResult(test.getTestName(), packageName, shouldSkip, failedOrSkippedTests,
+                                new Error("The provided data set is empty."));
+                    } else {
+                        // Handle array data sets
+                        Class<?>[] argTypes = extractArgumentTypes((BArray) valueSets);
+                        List<Object[]> argList = extractArguments((BArray) valueSets);
+                        for (int i = 0, argListSize = argList.size(); i < argListSize; i++) {
+                            invokeDataDrivenTest(suite, test.getTestName(), String.valueOf(i), classLoader, scheduler,
+                                    shouldSkip, packageName, argList.get(i), argTypes, failedOrSkippedTests);
+                        }
                     }
                 } else if (valueSets instanceof BError || valueSets instanceof Error ||
```

Comment:
Can't we remove this condition? `valueSets instanceof BError`
I think it'scovered by this `valueSets instanceof Exception`

Reply:
Fixed with https://github.com/Dilhasha/plugin-vscode.git